prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,205 @@
+        name=job_name, namespace=namespace, **kube_kwargs
+    )
+
+
+@task
+async def list_namespaced_job(
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1JobList:
+    """Task for listing namespaced Kubernetes jobs.
+
+    Args:
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        namespace: The Kubernetes namespace to list jobs from.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+    Returns:
+        A Kubernetes `V1JobList` object.
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    kube_kwargs = kube_kwargs or {}
+
+    return api_client.list_namespaced_job(
+        namespace=namespace,
+        **kube_kwargs,
+    )
+
+
+@task
+async def patch_namespaced_job(
+    job_name: str,
+    body: Dict,
+    kubernetes_credentials: KubernetesCredentials = None,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for deleting a namespaced Kubernetes job.
+
+    Args:
+        job_name: The name of a job to patch.
+        body: A dictionary representation of a Kubernetes V1Job
+            specification.
+        namespace: The Kubernetes namespace to patch this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+    Raises:
+        ValueError: if `job_name` is `None`
+
+    Returns:
+        A Kubernetes `V1Job` object.
+    """
+
+    api_client = kubernetes_credentials.get_batch_client()
+
+    kube_kwargs = kube_kwargs or {}
+
+    return api_client.patch_namespaced_job(
+        name=job_name, namespace=namespace, body=body, **kube_kwargs
+    )
+
+
+@task
+async def read_namespaced_job(
+    job_name: str,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for reading a namespaced kubernetes job.
+
+    Args:
+        job_name: The name of a job to read.
+        namespace: The Kubernetes namespace to read this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+    Raises:
+        ValueError: if `job_name` is `None`
+
+    Returns:
+        A Kubernetes `V1Job` object.
+    """
+    if not job_name:
+        raise ValueError("The name of a Kubernetes job must be provided.")
```

Comment:
I don't think we need this since it's a required arg, unless this is to check empty names.